mmaction/datasets/base.py

```diff
@@ -90,19 +92,83 @@ def load_json_annotations(self):
                 video_infos[i]['label'] = video_infos[i]['label'][0]
         return video_infos
 
-    @abstractmethod
-    def evaluate(self, results, metrics, logger):
-        """Evaluation for the dataset.
+    def evaluate(self,
+                 results,
+                 metrics='top_k_accuracy',
+                 topk=(1, 5),
```

Comment:
putting topk at base dataset is not very good

Reply:
agree. topk is a option for 'top_k_accuracy'. I think a better way is to warp metrics as a class, and use config dict to initialize it like
```python
metrics = build_metrics(dict(type='top_k_accuracy'), topk=(1, 5))
```